clamav.py

```diff
@@ -205,6 +206,36 @@ def scan_file(path):
         signature = summary.get(path, AV_SIGNATURE_UNKNOWN)
         return AV_STATUS_INFECTED, signature
     else:
-        msg = "Unexpected exit code from clamscan: %s.\n" % av_proc.returncode
+        msg = "Unexpected exit code from clamdscan: %s.\n" % av_proc.returncode
         print(msg)
         raise Exception(msg)
+
+def start_clamd_daemon():
+    s3 = boto3.resource("s3")
+    s3_client = boto3.client("s3")
+
+    to_download = update_defs_from_s3(
+        s3_client, AV_DEFINITION_S3_BUCKET, AV_DEFINITION_S3_PREFIX
+    )
+
+    for download in to_download.values():
+        s3_path = download["s3_path"]
+        local_path = download["local_path"]
+        print("Downloading definition file %s from s3://%s" % (local_path, s3_path))
+        s3.Bucket(AV_DEFINITION_S3_BUCKET).download_file(s3_path, local_path)
+        print("Downloading definition file %s complete!" % (local_path))
+
+    av_env = os.environ.copy()
+    av_env["LD_LIBRARY_PATH"] = CLAMAVLIB_PATH
+
+    print("Starting clamd")
+
+    subprocess.Popen(
+        ["%s/clamd" % CLAMAVLIB_PATH, "-c", "%s/scan.conf" % CLAMAVLIB_PATH],
+        stderr=subprocess.STDOUT,
+        stdout=subprocess.PIPE,
+        env=av_env,
+    ).wait()
+
+    clamd_log_file = open("/tmp/clamd.log")
+    print(clamd_log_file.read())
```

Comment:
Is this print to std out just so we know things are running?
I'm not sure we truncate it, so we could end up printing a large log file here.

Reply:
> Is this print to std out just so we know things are running?

Correct

> I'm not sure we truncate it, so we could end up printing a large log file here.

The startup log is quite short. Here's the full output:
```
+++ Started at Tue Feb 11 00:47:51 2020
Received 0 file descriptor(s) from systemd.
clamd daemon 0.101.5 (OS: linux-gnu, ARCH: x86_64, CPU: x86_64)
Log file size limited to 1048576 bytes.
Reading databases from /tmp/clamav_defs
Not loading PUA signatures.
Bytecode: Security mode set to "TrustSigned".
Loaded 6736784 signatures.
LOCAL: Unix socket file /tmp/clamd.sock
LOCAL: Setting connection queue length to 200
Limits: Global time limit set to 120000 milliseconds.
Limits: Global size limit set to 104857600 bytes.
Limits: File size limit set to 26214400 bytes.
Limits: Recursion level limit set to 16.
Limits: Files limit set to 10000.
Limits: MaxEmbeddedPE limit set to 10485760 bytes.
Limits: MaxHTMLNormalize limit set to 10485760 bytes.
Limits: MaxHTMLNoTags limit set to 2097152 bytes.
Limits: MaxScriptNormalize limit set to 5242880 bytes.
Limits: MaxZipTypeRcg limit set to 1048576 bytes.
Limits: MaxPartitions limit set to 50.
Limits: MaxIconsPE limit set to 100.
Limits: MaxRecHWP3 limit set to 16.
Limits: PCREMatchLimit limit set to 100000.
Limits: PCRERecMatchLimit limit set to 2000.
Limits: PCREMaxFileSize limit set to 26214400.
Archive support enabled.
AlertExceedsMax heuristic detection disabled.
Heuristic alerts enabled.
Portable Executable support enabled.
ELF support enabled.
Mail files support enabled.
OLE2 support enabled.
PDF support enabled.
SWF support enabled.
HTML support enabled.
XMLDOCS support enabled.
HWP3 support enabled.
Self checking every 600 seconds.
```